pkg/grpcfactory/interceptors_test.go

```diff
@@ -39,7 +37,7 @@ var _ = Describe("gRPC Factory Interceptors", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 
-		It("should handle invocation error and log it", func() {
+		It("should handle invocation error and log it", func(ctx SpecContext) {
```

Comment:
For my own understanding, what is adding the `ctx` here doing? Is it just for consistency?

Reply:
Not needed indeed.
It was a find and replace 😓

Reply:
My bad, I thought it was for another test, github hid the usage
I thought the usage is not there but I am actually using it.
Yes I am using it for `err := grpcfactory.ApiLogger(ctx, method, req, reply, cc, invoker)`

Adding ctx here primarily ensures that the test works within Ginkgo’s managed context lifecycle. This allows the invoker to receive a properly managed context.Context, which supports cancellation and timeout handling. Even if this particular test doesn’t leverage these features, including ctx ensures consistency with real-world function signatures and enables future flexibility (e.g., if the invoker uses context-sensitive logic like deadlines or tracing).

Yes, I agree, I actually started `specContext` following this after the comment from @jonathan-gramain in another PR. But using a context helps anyways, whether its `context.TODO` or `context.Background`

Reply:
that said we do need a styling guide for GO indeed